addon/services/infinity.js

```diff
@@ -113,18 +142,169 @@ export default Service.extend({
 
   /**
     Trigger a load of the next page of results while also checking if it can load more
+    Subsequent fetching.  Not used for initial request
 
     @public
     @method infinityLoad
     @param {Ember.ArrayProxy} infinityModel
     @param {Integer} increment - to increase page by 1 or -1
+    @param {Node} infinityLoaderElem - important when have multiple infinity loader elements on the page
    */
-  infinityLoad(infinityModel, increment) {
-    if (get(infinityModel, '_loadingMore') || !get(infinityModel, '_canLoadMore')) {
+  infinityLoad(infinityModel, increment = 1, infinityLoaderElem) {
```

Comment:
I still don't understand why we're using `infinityLoaderElem` here. Passing around the element is brittle (re-renders aren't always guaranteed to be the same "in memory" object in single page app frameworks)

Services in Ember shouldn't really have anything to do with display logic or the DOM at all, right?

Reply:
That is a good point.  And now that you bring it up, that is a good point.  

Every time we ask for a new set of records, there are two things that are motivating passing around the elem that called this load method

1.  Multiple loader elements on the page 
2.  Need to keep calling (recursively) to load the page with elements until the loader elem is off the page.

But I’ll look into decoupling the service from the loader elem.

Reply:
1. Multiple loader elements on the page

The infinity-loader still needs a way to receive a reference at an infinityModel from the route, it could be the infinityModel itself, or a cache key, or whatever.

Once it has a reference to the infinityModel it cares about, it should only respond to changes to that specific model, meaning you can have as many on the page as you want, without worrying about the specific DOM element that represents it.

2. Need to keep calling (recursively) to load the page with elements until the loader elem is off the page.

See above!